src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,8 +1,43 @@
-import {Component} from '@angular/core';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Product} from 'src/app/shared/products/product.interface';
+
+const MAX_RATING = 5;
+
+interface RatingArr {
+    active: boolean;
+}
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
-export class CardComponent {}
+export class CardComponent {
+    @Input() product: Product | null = null;
+
+    @Output() buyProduct: EventEmitter<Product | null> = new EventEmitter();
```

Comment:
Не плохо было бы пометить свойство как `readonly`
Почему решили указать явную типизацию, какие видишь преимущества данного подхода?

Мне тут видится тавтология, почему бы не указать так:
```suggestion
    @Output() readonly buyProduct = new EventEmitter<Product | null>();
```